src/main/kotlin/com/londogard/summarize/summarizers/EmbeddingClusterSummarizer.kt

```diff
@@ -1,3 +1,166 @@
 package com.londogard.summarize.summarizers
 
-// TODO implement this
\ No newline at end of file
+import com.londogard.smile.SmileOperators
+import com.londogard.smile.extensions.*
+import com.londogard.summarize.mutableSumByCols
+import com.londogard.summarize.embeddings.WordEmbeddings
+import com.londogard.summarize.extensions.normalize
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+enum class ScoringConfig {
+    Rosselio,
+    Ghalandari
+}
+
+internal class EmbeddingClusterSummarizer(
+    val threshold: Double = 0.3,
+    val simThreshold: Double = 0.95,
+    val config: ScoringConfig = ScoringConfig.Ghalandari
+) : SmileOperators, SummarizerModel {
+    private val embeddings: WordEmbeddings = WordEmbeddings(dimensions = 50)
+    private val zeroArray = Array(embeddings.dimensions) { 0f }
+    private fun String.simplify(): String =
+        normalize().toLowerCase().words().joinToString(" ")
+
+    private fun getTfIdf(sentences: List<String>): Pair<List<Double>, Map<Int, String>> {
+        val corpus = sentences.map { it.bag(stemmer = null) }
+        val words = corpus.flatMap { bag -> bag.keys }.distinct()
+        val bags = corpus.map {
```

Comment:
one-line & put named lambda. 
Finally, it's correct as the words vectorizing the sentence is the full corpora